sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverClient.java

```diff
@@ -41,8 +42,11 @@
         .setIsAutoCompleteEnabled(false)
         .setMaxAutoLockRenewalDuration(Duration.ZERO);
 
-    private final AtomicReference<EmitterProcessor<ServiceBusReceivedMessageContext>> messageProcessor =
+    private final AtomicReference<EmitterProcessor<ServiceBusReceivedMessageContext>> messageSource =
         new AtomicReference<>();
+    private final AtomicReference<EmitterProcessor<SynchronousMessageSubscriber>> workQueueProcessor =
```

Comment:
Why do you need another emitter processor? Processors are useful when it listens upstream and publishes downstream. In this case, you are using it as a sink.